common_logging/src/main/java/uk/gov/dhsc/htbhf/logging/event/EventFailedEvent.java

```diff
@@ -0,0 +1,28 @@
+package uk.gov.dhsc.htbhf.logging.event;
+
+import lombok.Builder;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Represents an event that has failed.
+ */
+public class EventFailedEvent extends Event {
+
+    public static final String FAILED_EVENT_KEY = "failedEvent";
+    public static final String FAILURE_DESCRIPTION_KEY = "failureDescription";
+
+    @Builder
+    public EventFailedEvent(String failureDescription, Event failedEvent) {
+        super(CommonEventType.FAILED_EVENT, failedEvent.getTimestamp(), constructMetadata(failureDescription, failedEvent));
+    }
+
+    private static Map<String, Object> constructMetadata(String failureDescription, Event failedEvent) {
+        Map<String, Object> metadata = new HashMap<>(failedEvent.getEventMetadata());
+        metadata.put(FAILED_EVENT_KEY, failedEvent.getEventType());
```

Comment:
Do we also want to store the timestamp from the original event?

Reply:
we're using the original timestamp as the timestamp for the failure event